src/pdm4ar/exercises/ex04/structures.py

```diff
@@ -25,9 +25,13 @@ class Cell(IntEnum):
     START = 1
     GRASS = 2
     SWAMP = 3
+    WORMHOLE = 4
+    CLIFF = 5
 
 
 Policy = NDArray[np.int64]
 """Type Alias for the policy"""
+AllOptimalActions = Union[NDArray[np.object_], Policy]
```

Comment:
Is this to keep in the same object the (possibly) multiple optimal actions?

Reply:
It is an `NDArra`y containing either a list of optimal actions or a single action. Each entry of `NDArray[np.object_]` points to a list object, and I perform dynamic type-checking during usage. 

The design allows for flexibility in handling either a list or a single action as ground truth optimal actions for evaluation. Adding the type `Policy`  is mainly for backward compatibility as the old ground truth optimal actions have type `Policy`. However, since it should support a list of actions for generality, we can define AllOptimalActions = NDArray[np.object_] exclusively.

Reply:
Now, all ground truth policies have the type `NDArray[np.object_]`. We can safely define `AllOptimalActions = NDArray[np.object_]` exclusively.

Reply:
yep, I see. I would simply add a comment underneath to better explain it to the students.
If `Policy` is not used anymore feel free to remove to avoid confusion for the students.
Regarding the name I would change `AllOptimalActions` to simply `OptimalActions`.